keychain with md5 doesn't work on newer Linux

See on latest master as of May 3, 2024  (git sha 53820a5753)

The topotest test_keychain fails on Debian 12 and Ubuntu 24.04 (It does work on Debian 10 and Ubuntu 18.04, unknown on others):

```
test_keychain: test_send_accept
AssertionError: assert False
 +  where False = is_key_active(<lib.topogen.TopoRouter object at 0x7fb3f89a50>, 'kc', 1, 0)
E   AssertionError: assert False
     +  where False = is_key_active(<lib.topogen.TopoRouter object at 0x7fb3f89a50>, 'kc', 1, 0)
```

The issue can be seen in the router logs of the topotest:
```
r1# conf t
r1(config)# key chain kc
r1(config-keychain)#  key 1
r1(config-keychain-key)#   key-string theSecret
r1(config-keychain-key)#   crypto-algorithm md5
% Unknown command:   crypto-algorithm md5
r1(config-keychain-key)#  exit
r1(config-keychain)# exit
r1(config)#
```

My guess (not verified) is that the lib used doesn't support md5 anymore on newer systems.

--- a/tests/topotests/key_sendaccept/test_keychain.py
+++ b/tests/topotests/key_sendaccept/test_keychain.py
@@ -62,7 +62,7 @@ def test_send_accept(tgen):
 key chain kc
  key 1
   key-string theSecret
-  crypto-algorithm md5
+  cryptographic-algorithm hmac-sha-256
  exit
 exit
 """
@@ -74,7 +74,7 @@ exit
 key chain kc
  key 1
   key-string theSecret
-  crypto-algorithm md5
+  cryptographic-algorithm hmac-sha-256
   send-lifetime 00:00:00 Jan 1 2024 infinite
   accept-lifetime 00:00:00 Jan 1 2024 infinite
  exit
@@ -101,7 +101,7 @@ exit
 key chain kc
  key 2
   key-string theSecret
-  crypto-algorithm md5
+  cryptographic-algorithm hmac-sha-256
   send-lifetime 00:00:00 Jan 1 2024 duration {secs_in_10_years}
   accept-lifetime 00:00:00 Jan 1 2024 duration {secs_in_10_years}
  exit
@@ -127,7 +127,7 @@ exit
 key chain kc
  key 3
   key-string theSecret
-  crypto-algorithm md5
+  cryptographic-algorithm hmac-sha-256
   send-lifetime   00:00:00 Jan 1 2024  23:59:59 Dec 31 2034
   accept-lifetime 00:00:00 Jan 1 2024  23:59:59 Dec 31 2034
  exit

